File: src/users/entities.py
from dataclasses import dataclass
from datetime import datetime
from uuid import UUID


@dataclass(kw_only=True)
class UserFields:
    username: str
    email: str
    email_verified: bool
    image_url: str | None = None
    hashed_password: bytes
    active: bool
    created_at: datetime


@dataclass(kw_only=True)
class User(UserFields):
    id: UUID

File: client/src/entities.ts

export type User = {
    id: string
    username: string
    email: string
    email_verified: boolean
    image_url: string | null
    active: boolean
    created_at: Date
}


export type ValidationError = {
    code: number;
    message: string;
    errors: {
        [key: string]: {
            code: string;
            message: string;
        }
    }
};

export class ServiceError extends Error {
    error: ValidationError
    constructor(error: ValidationError) {
        super(error.message);
        this.error = error;
    }
}
